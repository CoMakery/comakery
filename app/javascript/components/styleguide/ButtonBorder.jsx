import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Button from './Button'

class ButtonBorder extends React.Component {
  render() {
    const {className, ...other} = this.props

    const classnames = classNames(
      'button__border',
      className
    )

    return (
      <React.Fragment>
        <Button className={classnames} {...other} />
      </React.Fragment>
    )
  }
}

ButtonBorder.propTypes = {
  className: PropTypes.string
}
ButtonBorder.defaultProps = {
  className: ''
}
export default ButtonBorder
