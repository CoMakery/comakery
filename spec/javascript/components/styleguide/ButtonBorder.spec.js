import React from 'react'
import { shallow, mount, render } from 'enzyme'
import ButtonBorder from 'components/styleguide/ButtonBorder'

describe('ButtonBorder', () => {
  it('renders correctly without props', () => {
    const wrapper = shallow(<ButtonBorder />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.exists('.button__border')).toBe(true)
  })

  it('renders correctly with custom className', () => {
    const wrapper = shallow(<ButtonBorder className="__test" />)

    expect(wrapper.exists('.button__border.__test')).toBe(true)
  })
})
