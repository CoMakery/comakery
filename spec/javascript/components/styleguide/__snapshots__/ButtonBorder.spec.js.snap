// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBorder renders correctly without props 1`] = `
<Fragment>
  <Button
    className="button__border"
    disabled={false}
    type="button"
    value="publish"
  />
</Fragment>
`;
